File: School-Management-System/WebApi/ClientApp/src/app/home/components/teacher/shared/models/viewModels/teacher.viewModel.ts
export interface TeacherViewModel {
  id: string;
  userId?: string;
  employeeCode?: string;
  firstName: string;
  middleName?: string;
  lastName: string;
  fullName: string;
  gender: number;
  age?: number;
  dateOfBirthNp?: string;
  dateOfBirthEn?: string;
  contactNumber: string;
  alternateContactNumber?: string;
  email?: string;
  provinceId?: number;
  districtId?: number;
  municipalityId?: number;
  wardNo?: number;
  fatherName?: string;
  motherName?: string;
  designation?: string;
  joiningDateNp?: string;
  joiningDateEn?: string;
  isActive: boolean;
  inactiveReason?: string;
  qualifications: TeacherQualificationViewModel[];
  experiences: TeacherExperienceViewModel[];
  documents: TeacherDocumentViewModel[];
  assignments: TeacherClassSectionViewModel[];
}

export interface TeacherQualificationViewModel {
  id: string;
  degreeName: string;
  institutionName: string;
  boardOrUniversity?: string;
  passedYear?: string;
  gradeOrPercentage?: string;
  majorSubject?: string;
  remarks?: string;
}

export interface TeacherExperienceViewModel {
  id: string;
  organizationName: string;
  designation: string;
  subjectOrDepartment?: string;
  startDateNp?: string;
  startDateEn?: string;
  endDateNp?: string;
  endDateEn?: string;
  isCurrent: boolean;
  remarks?: string;
}

export interface TeacherClassSectionViewModel {
  id: string;
  academicYearId: string;
  academicYearName: string;
  classSectionId: string;
  classRoomId: string;
  classRoomName: string;
  sectionId: string;
  sectionName: string;
  courseId: string;
  courseName: string;
  isClassTeacher: boolean;
  isActive: boolean;
  remarks?: string;
}

export interface TeacherDocumentViewModel {
  id: string;
  documentType: string;
  documentTitle: string;
  filePath: string;
  originalFileName: string;
  mimeType: string;
  fileSize: number;
  uploadedDate: string;
}

export interface TeacherDashboardViewModel {
  totalActiveTeachers: number;
  totalInactiveTeachers: number;
  assignedTeachers: number;
  classTeachers: number;
}

File: School-Management-System/Application/Teachers/TeacherViewModel.cs
namespace Application.Teachers
{
    public class TeacherViewModel
    {
        public Guid Id { get; set; }
        public Guid? UserId { get; set; }
        public string? EmployeeCode { get; set; }
        public string FirstName { get; set; }
        public string? MiddleName { get; set; }
        public string LastName { get; set; }
        public string FullName { get; set; }
        public int Gender { get; set; }
        public int? Age { get; set; }
        public string? DateOfBirthNp { get; set; }
        public string? DateOfBirthEn { get; set; }
        public string ContactNumber { get; set; }
        public string? AlternateContactNumber { get; set; }
        public string? Email { get; set; }
        public int? ProvinceId { get; set; }
        public int? DistrictId { get; set; }
        public int? MunicipalityId { get; set; }
        public int? WardNo { get; set; }
        public string? FatherName { get; set; }
        public string? MotherName { get; set; }
        public string? Designation { get; set; }
        public string? JoiningDateNp { get; set; }
        public string? JoiningDateEn { get; set; }
        public bool IsActive { get; set; }
        public string? InactiveReason { get; set; }
        public List<TeacherQualificationViewModel> Qualifications { get; set; } = new();
        public List<TeacherExperienceViewModel> Experiences { get; set; } = new();
        public List<TeacherDocumentViewModel> Documents { get; set; } = new();
        public List<TeacherClassSectionViewModel> Assignments { get; set; } = new();
    }

    public class TeacherQualificationViewModel
    {
        public Guid Id { get; set; }
        public string DegreeName { get; set; }
        public string InstitutionName { get; set; }
        public string? BoardOrUniversity { get; set; }
        public string? PassedYear { get; set; }
        public string? GradeOrPercentage { get; set; }
        public string? MajorSubject { get; set; }
        public string? Remarks { get; set; }
    }

    public class TeacherExperienceViewModel
    {
        public Guid Id { get; set; }
        public string OrganizationName { get; set; }
        public string Designation { get; set; }
        public string? SubjectOrDepartment { get; set; }
        public string? StartDateNp { get; set; }
        public string? StartDateEn { get; set; }
        public string? EndDateNp { get; set; }
        public string? EndDateEn { get; set; }
        public bool IsCurrent { get; set; }
        public string? Remarks { get; set; }
    }

    public class TeacherClassSectionViewModel
    {
        public Guid Id { get; set; }
        public Guid AcademicYearId { get; set; }
        public string AcademicYearName { get; set; }
        public Guid ClassSectionId { get; set; }
        public Guid ClassRoomId { get; set; }
        public string ClassRoomName { get; set; }
        public Guid SectionId { get; set; }
        public string SectionName { get; set; }
        public Guid CourseId { get; set; }
        public string CourseName { get; set; }
        public bool IsClassTeacher { get; set; }
        public bool IsActive { get; set; }
        public string? Remarks { get; set; }
    }

    public class TeacherDocumentViewModel
    {
        public Guid Id { get; set; }
        public string DocumentType { get; set; }
        public string DocumentTitle { get; set; }
        public string FilePath { get; set; }
        public string OriginalFileName { get; set; }
        public string MimeType { get; set; }
        public long FileSize { get; set; }
        public DateTime UploadedDate { get; set; }
    }

    public class TeacherDashboardViewModel
    {
        public int TotalActiveTeachers { get; set; }
        public int TotalInactiveTeachers { get; set; }
        public int AssignedTeachers { get; set; }
        public int ClassTeachers { get; set; }
    }

    public class TeacherAccountViewModel
    {
        public bool IsAccountCreated { get; set; }
        public Guid? UserId { get; set; }
        public string? UserName { get; set; }
        public string? Email { get; set; }
        public string? PhoneNumber { get; set; }
        public bool IsActive { get; set; }
        public List<string> Roles { get; set; } = new();
    }
}
